### ISSUE TITLE
Gtk: Invalidating a TreeGridView with a CheckBoxCell does not redraw the CheckBox

### ISSUE THREAD

**Issue by Sejsel** [picoe/Eto#1982]
## Expected Behavior
Calling `Invalidate()` should queue a redraw for the `TreeGridView` and its children on the GTK platform (as the documentation claims it would)

## Actual Behavior
Checkboxes do not get redrawn after being set to a different value, hovering over them is necessary.

## Steps to Reproduce the Problem

  1. Create a TreeGridView with a CheckBoxCell column bound to a bool property
  2. Add some rows
  3. Set all rows to checked
  4. Call Invalidate on the TreeGridView

## Code that Demonstrates the Problem

I currently don't have a simple sample, will try to add one later.

## Specifications

  - Version: 2.5.11
  - Platform(s): Gtk3
  - Operating System(s): Arch Linux, gtk3 1:3.24.29-2

**Comment by cwensley** [comment]
Hi @Sejsel, thanks for reporting the issue!

For this you should use `TreeGridView.ReloadItem()` or `ReloadData()`.  `Invalidate()` simply repaints the current state.

Hope this helps!  Feel free to comment/reopen if the above does not work.

**Comment by Sejsel** [comment]
> Hope this helps! Feel free to comment/reopen if the above does not work.

Thank you, it somewhat does, but is blocked by another issue. Interestingly enough, invalidating works on WPF for this.

Reloading is what I already tried, but sadly the reloads reset the state of the GridView on GTK (scrolling to the top, collapsing sections) - it's tracked as #1290. My current workaround is to swap to a different tab and back to trigger a full repaint. I can only do that because it happens to be in a tab in my program, and miraculously it doesn't have any negative effects, but it probably would in a more complex program.

**Comment by cwensley** [comment]
Hey @Sejsel, thanks for the feedback on the issue.  Sounds like at least something is needed for your scenario on Gtk, I'll reopen this and take a look.

**Comment by cwensley** [comment]
Closing this as Invalidate() shouldn't be used to notify the GridView/TreeGridView that the item state has changed.  One should use ReloadItem/ReloadData instead with the particular item/row.
